### PR TITLE
Fix multiple STIG IDs for RHEL8

### PR DESCRIPTION
**Pull request by nightmared**
Hello,

this adds/replace multiple STIG identifiers that are missing in the current STIG profile.

### CODE REVIEW

**Comment by nightmared** [comment]
Should I update the profile stability?

File: linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ references:
     pcidss4: "1.5.1"
     srg: SRG-OS-000480-GPOS-00227
     stigid@rhel7: RHEL-07-040810
+    stigid@rhel8: RHEL-08-040090
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml:53]: That is the reasoning behind this change?

If you want automation for this rule I think we should add it to `configured_firewalld_default_deny` verses changing the mapping. That can be done in another PR.

File: products/rhel8/profiles/stig.profile [hunk 2]
```diff
@@ -641,7 +650,7 @@ selections:
     - account_disable_post_pw_expiration
 
     # RHEL-08-020270
-    - account_temp_expire_date
+    - account_emergency_expire_date
```
Comment (Mab879) [inline products/rhel8/profiles/stig.profile:653]: What is the reason for this change? 

The two rules are very similar but the latest STIGs due use the temporary wording vs emergency. So I think that the existing mapping is fine.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR. I have a couple of comments.

Please update the profile stability once you are done as well.

File: products/rhel8/profiles/stig.profile (hunk 2 above)
Comment (nightmared) [reply products/rhel8/profiles/stig.profile:653]: After re-reading more closely the STIG guidelines, I believe you are right ("Emergency accounts are not subject to manual removal or scheduled expiration requirements.").

However, that also means we should assign RHEL-08-020270 to account_temp_expire_date, and remove that same STIG reference in account_emergency_expire_date, right?
I will do that in the next version of this PR.

File: linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml (hunk 1 above)
Comment (nightmared) [reply linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml:53]: To be honest, I am unsure about the distinctions between both rules.
If I'm not mistaken, you draw the distinction with:
- `configured_firewalld_default_deny` approach is to update the zone of the existing interfaces
- `set_firewalld_default_zone` approach is to update the firewalld.conf file to contain `DefaultZone=drop`

Is that understanding right?

If so, perhaps [my other PR](https://github.com/ComplianceAsCode/content/pull/11251) should be split between `configured_firewalld_default_deny` and `set_firewalld_default_zone`. Additionally, a check (probably with SCE because there is no firewalld backend in OpenSCAP to query live interfaces, AFAIK) should be written for configured_firewalld_default_deny.
Finally, as you suggested, the STIG identifier should be present on both rules.

Does that course of action appear reasonable?

File: products/rhel8/profiles/stig.profile (hunk 2 above)
Comment (Mab879) [reply products/rhel8/profiles/stig.profile:653]: > However, that also means we should assign RHEL-08-020270 to account_temp_expire_date, and remove that same STIG reference in account_emergency_expire_date, right? I will do that in the next version of this PR.

That sounds reasonable.

File: linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml (hunk 1 above)
Comment (Mab879) [reply linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml:53]: Sounds reasonable to me. 

I will note that the data streams in RHEL and the releases in this project don't ship with SCE content.

Comment (nightmared) [reply linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml:53]: After thinking about this a bit further, I think adding a remediation "by default" for these rules is quite dangerous, as the warning in https://github.com/ComplianceAsCode/content/blob/8c3e9d6c127522e28d9ed7010476139c01fd8d37/linux_os/guide/system/network/network-firewalld/ruleset_modifications/set_firewalld_default_zone/rule.yml#L62-L67 rightly indicates. Given the risks involved, adding remediations for them seems a bit reckless on my part.

For now, I will simply enable both rules in the stig profile, with `configured_firewalld_default_deny` having no test/remediation, and `set_firewalld_default_zone ` being tested but not remediated.

I will circle back with colleagues next week to determine what could be a proper course of action for this specific pair of rules.

As for the lack of SCE content in official RHEL releases of CAC, this is not a problem for us as we apply multiple configuration on top of the STIG profile and thus were already rebuilding CAC. However, that will indeed be a limitation of said remediation, should we agree on going toward that route.

**Review by Mab879 — APPROVED** [review]
LGTM.

Thanks!

**Comment by Mab879** [comment]
Waving Auomatus tests as they pass locally.

```
$ ./automatus.py rule --datastream ../build/ssg-rhel8-ds.xml --libvirt qemu:///system automatus_rhel8_8 set_firewalld_default_zone
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/Developer/ComplianceAsCode/content/tests/logs/rule-custom-2023-11-08-1428/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_set_firewalld_default_zone
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script wrong.fail.sh using profile (all) OK
```
